org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/ScalaProject.scala

```diff
@@ -67,15 +71,27 @@ object ScalaProject {
 
   /** Listen for [[IWorkbenchPart]] event and takes care of loading/discarding scala compilation units.*/
   private class ProjectPartListener(project: ScalaProject) extends PartAdapter with HasLogger {
+    override def partActivated(part: IWorkbenchPart): Unit = {
```

Comment:
yes, the PC will be restarted on every focus change if there are errors in the editor that is focused. I know this can hardly be described as a clean solution but I couldn't think of a better one. At least, the behavior is optional and can be disabled in the preferences.